added BoundaryMeshElementMeasure

jantomec opened the pull request:
Added function **BoundaryMeshElementMeasure** which calculates the area of boundary element mesh object or the boundary area of element mesh object.

jantomec committed:
added BoundaryMeshElementMeasure

Added function BoundaryMeshElementMeasure which calculates the area of boundary element mesh object or the boundary area of element mesh object.

PintarM commented on MeshTools.wl:
```diff
@@ -150,15 +150,63 @@ MeshElementMeasure[mesh_ElementMesh]:=Module[{
 (*BoundaryElementMeasure*)
 
 
+Clear[elementMeasure]
```
Typo! I think it should be Clear[boundaryElementMeasure]

PintarM commented on MeshTools.wl:
```diff
@@ -150,15 +150,63 @@ MeshElementMeasure[mesh_ElementMesh]:=Module[{
+	type_,
+	order_,
+	integrationOrder_]:=
+	Total[boundaryElementMeasure[#,type,order,integrationOrder]&/@nodes]
```
Use of Total[...] breaks the intended functionality. We would like to return "BoundaryElement" measure element by element.

PintarM commented on MeshTools.wl:
```diff
@@ -150,15 +150,63 @@ MeshElementMeasure[mesh_ElementMesh]:=Module[{
+(* Boundary mesh measure for each 2D element. *)
+boundaryElementMeasure[
+	nodes_List/;(Depth[nodes]==3),
+	type_/;(type==TriangleElement||type==QuadElement),
```
I think this condition is redundant (for function in the private context).

PintarM commented on MeshTools.wl:
```diff
@@ -150,15 +150,63 @@ MeshElementMeasure[mesh_ElementMesh]:=Module[{
-	{},
-	Print["Placeholder"];
-]
+BoundaryMeshElementMeasure[mesh_ElementMesh,integrationOrder_:3]:=Module[
```
I prefer the original name BoundaryElementMeasure. ElementMesh object has three types of elements ("MeshElements", "BoundaryElements" and "Pointelements") and my naming logic follows this.

I think integrationOrder argument is redundant and should be automatically inferred from "MeshOrder"

jantomec committed:
BoundaryElementMeasure enhancement

Changed BoundaryElementMeasure function so that it returns a list of each elements' area. Added test examples

PintarM committed:
Merge branch 'master' into master
